CMakeLists.txt

```diff
@@ -75,6 +77,14 @@ endif()
 if(ACTS_BUILD_DOC)
   find_package(Doxygen 1.8.11 REQUIRED)
 endif()
+if(ACTS_BUILD_CUDA_FEATURES)
+  add_definitions(-DACTS_HAS_CUDA) 
+  enable_language(CUDA)
+  set(CMAKE_CUDA_STANDARD 14)
+  set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+  find_library(CUDART_LIBRARY cudart ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+  include_directories(${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+endif()
```

Comment:
Ok this sets a lot of global parameters. We usually try not to do that. My preferred approach here would be for you to create a 'CUDAPlugin' which is a separate target from the Core target. All these parameters, I believe, you should be able to set only on that new target. This way, you don't influence the compilation of the Core target. This may require you to separate the CUDA implementation of the seedfinder into this separate target, since the CPU seedfinder sits in the Core.

We can discuss this more if it's not clear.

Reply:
Now there are only three lines
```
  enable_language(CUDA) 
  set(CMAKE_CUDA_STANDARD 14) 
  set(CMAKE_CUDA_STANDARD_REQUIRED ON)
```
Do you think it is still too much? How about moving some flag options to `ActsCompilerOptions.cmake`?